### DISCUSSION TITLE
The configuration pageExtensions doesn't work when using `app/` directory (v13.x.x)

### DISCUSSION THREAD

**Discussion by ningo-agilityio** [vercel/next.js#42883] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.4.0: Mon Feb 21 20:36:53 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T8101
    Binaries:
      Node: 18.11.0
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 7.6.0
    Relevant packages:
      next: 13.0.2
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0

warn  - Latest canary version not detected, detected: "13.0.2", newest: "13.0.3-canary.1".

### What browser are you using? (if relevant)

Chrome 107.0.5304.87

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I have just investigated time on the latest version v13.x.x, I'm using concurrently both folder `app/` and  `pages` then I faced this issue when I tried to configure pageExtensions in `next.config.js`

```
TypeError: pageExtensions.map is not a function
```

My file `next.config.js`:

```
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  experimental: { appDir: true },
  pageExtensions: ['page.tsx', 'page.ts', 'page.jsx', 'page.js'],
}
```

### Expected Behavior

I expected that we can use pageExtensions for both app/ and pages/ directories.

### Link to reproduction

...

### To Reproduce

...

**Comment by Lrunlin** [comment]


**Reply by claudio-jeeves** [reply]
is this solving the issue? By not adding a dot into the extension (like "tsx" in your screenshot) we are getting a compile error

**Comment by mathieuguyot** [comment]
Indeed, this would be great !
I have hard time reading my file inspector with all pages.tsx that looks the same :| 
The worst thing is when I try to open a file lets say i'm working on a Activities page, It will be good to just search for ActivitiesPages.tsx or Activities.page.tsx or something that way !
